sympy/combinatorics/fp_groups.py

```diff
@@ -259,7 +259,7 @@ def _is_infinite(self):
         used_gens = set()
         for r in self.relators:
             used_gens.update(r.contains_generators())
-        if any([g not in used_gens for g in self.generators]):
+        if not all(g in used_gens for g in self.generators):
```

Comment:
I'm not sure what the type of `self.generators` is but `used_gens` is a set and this looks a subset test to me. The most efficient way would be:
```python
if not set(self.generators) <= used_gens:
```
It's not possible to negate the `<=` because this is a partial order.